java/src/jmri/jmrit/logixng/expressions/swing/FileAsFlagSwing.java

```diff
@@ -42,7 +43,7 @@ protected void createPanel(@CheckForNull Base object, @Nonnull JPanel buttonPane
         JPanel tabbedPaneNamedBean;
         JPanel tabbedPaneState;
 
-        tabbedPaneNamedBean = _selectFilenameSwing.createFilenamePanel(expression.getSelectFilename());
+        tabbedPaneNamedBean = _selectFilenameSwing.createFilenamePanel(expression.getSelectFilename(), FileUtil.getPreferencesPath());
```

Comment:
This needs to be getUserFilesPath.  getPreferencesPath is the `settings:` location.

Reply:
Thanks. Fixed. I wasn't sure of which one to use.